aurora/src/main/java/src/config/JMXConfig.java

```diff
@@ -0,0 +1,141 @@
+            LOGGER.error("Channel with prefix {} already exists", channelPrefix);
+            throw new IllegalArgumentException("Channel with this name already exists");
+        }
+
+        ChannelProperty channelProperty = new ChannelProperty();
+        channelProperty.setPort(Integer.parseInt(port));
+        channelProperty.setHost(host);
+
+        channelPropertyMap.put(channelPrefix, channelProperty);
+        channels.addChannel(channelPrefix, channelProperty);
+
+        this.writeBackUp();
+
+        LOGGER.debug("New channel created from JMX server with host {} and port {}"
+                , host, port);
+    }
+
+    @ManagedOperation
+    public void deleteChannel(String channelPrefix) {
+        LOGGER.info("Deleting channel {} from  JMX server", channelPrefix);
+        if (!channelPropertyMap.containsKey(channelPrefix)) {
+            LOGGER.error("Channel with prefix {} does not exists", channelPrefix);
+            throw new IllegalArgumentException("Channel with this name does not exists");
+        }
+
+        this.channelPropertyMap.remove(channelPrefix);
+
+        this.channels.deleteChannel(channelPrefix);
+
+        this.writeBackUp();
+    }
+
+    @ManagedOperation
+    public void editChannel(String channelPrefix, String host, String port) {
+        LOGGER.info("Editing channel {} from  JMX server", channelPrefix);
+        if (!channelPropertyMap.containsKey(channelPrefix)) {
+            LOGGER.error("Channel with prefix {} does not exists", channelPrefix);
+            throw new IllegalArgumentException("Channel with this name does not exists");
+        }
+
+        ChannelProperty channelProperty = this.channelPropertyMap.remove(channelPrefix);
+        channelProperty.setHost(host);
+        channelProperty.setPort(Integer.parseInt(port));
+
+        this.channelPropertyMap.put(channelPrefix, channelProperty);
+
+        this.channels.editChannel(channelPrefix, channelProperty);
+
+        this.writeBackUp();
+    }
+
+
+    protected void writeBackUp() {
+        LOGGER.info("Writing back-up to json");
+
+
+        ObjectWriter objectWriter = new ObjectMapper().writerWithDefaultPrettyPrinter();
+
+        try (Writer output = new OutputStreamWriter(new FileOutputStream(ApplicationDirectoryUtil.getConfigFile("channels.json")), UTF_8)) {
+
+            String jsonData = getStringFromMap(this.channelPropertyMap, objectWriter);
+
+            output.write(jsonData);
+
+            LOGGER.info("Back-up written successfully");
+        } catch (IOException e) {
+            LOGGER.error("Exception thrown during writing back-up");
+        }
+    }
+
+    private String getStringFromMap(Map<String, ChannelProperty> data, ObjectWriter objectWriter)
+            throws JsonProcessingException {
+        LOGGER.debug("In getStringFromMap private method");
+
+        return objectWriter.writeValueAsString(data);
+    }
+
+    private ConcurrentHashMap<String, ChannelProperty> readChannelsConfigsFromFile() {
```

Comment:
this method seems to be duplication to the one in ChannelManager, use only one of them